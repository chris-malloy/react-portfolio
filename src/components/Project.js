import React from 'react';
import PropTypes from 'prop-types';
import { Card, CardTitle } from 'react-materialize';

const Project = (props) => {

  return (
    <div >
      <Card className='small black-text '
        header={<CardTitle image={props.image || 'image.png'}>{props.title}</CardTitle>}
        key={1}
        actions={[<a key={props.title} href={props.href}>{props.link}</a>]}>
        {props.description}
      </Card>
    </div>
 )
};

Project.propTypes= {
  title: PropTypes.string.isRequired,
  href: PropTypes.string.isRequired,
  link: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired
};

export default Project;