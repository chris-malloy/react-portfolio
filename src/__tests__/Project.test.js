import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Project from '../components/Project';

configure({ adapter: new Adapter() });

describe('Project', () => {
  const props = {
    title: 'title',
    href: 'href',
    link: 'link',
    description: 'description'
  }
  const project = shallow(<Project {...props} />)
  it('renders correctly', () => {
    expect(project).toMatchSnapshot();
  });
});