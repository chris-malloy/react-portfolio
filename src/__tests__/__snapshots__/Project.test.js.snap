// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Project
    description="description"
    href="href"
    link="link"
    title="title"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Card
        actions={
                Array [
                        <a
                          href="href"
                >
                          link
                </a>,
                      ]
        }
        className="small black-text "
        header={
                <CardTitle
                        image="image.png"
                >
                        title
                </CardTitle>
        }
>
        description
</Card>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": "1",
      "nodeType": "function",
      "props": Object {
        "actions": Array [
          <a
            href="href"
>
            link
</a>,
        ],
        "children": "description",
        "className": "small black-text ",
        "header": <CardTitle
          image="image.png"
>
          title
</CardTitle>,
      },
      "ref": null,
      "rendered": "description",
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Card
          actions={
                    Array [
                              <a
                                href="href"
                    >
                                link
                    </a>,
                            ]
          }
          className="small black-text "
          header={
                    <CardTitle
                              image="image.png"
                    >
                              title
                    </CardTitle>
          }
>
          description
</Card>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "actions": Array [
            <a
              href="href"
>
              link
</a>,
          ],
          "children": "description",
          "className": "small black-text ",
          "header": <CardTitle
            image="image.png"
>
            title
</CardTitle>,
        },
        "ref": null,
        "rendered": "description",
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
